misc/docerina/src/main/java/org/ballerinalang/docgen/docs/BallerinaDocGenerator.java

```diff
@@ -694,7 +694,10 @@ private static Path getModuleDocPath(Path absolutePkgPath) throws IOException {
     }
 
     private static Path getAbsoluteModulePath(String sourceRoot, Path modulePath) {
-        return Paths.get(sourceRoot).resolve(ProjectDirConstants.SOURCE_DIR_NAME)
-                               .resolve(modulePath);
+        Path sourcePath = Paths.get(sourceRoot);
+        if (sourcePath.endsWith(modulePath)) {
+            return Paths.get(sourceRoot);
+        }
+        return sourcePath.resolve(ProjectDirConstants.MODULES_ROOT).resolve(modulePath);
```

Comment:
@shehan360 , Could you check this code block ?

Reply:
Seems okay